test/sanity/code-smell/deprecated-config.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# (c) 2018, Matt Martz <matt@sivel.net>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import argparse
+import mmap
+import os
+import re
+
+import yaml
+
+from distutils.version import StrictVersion
+
+import ansible.config
+
+from ansible.plugins.loader import fragment_loader
+from ansible.release import __version__ as ansible_version
+from ansible.utils.plugin_docs import get_docstring
+
+DOC_RE = re.compile(b'^DOCUMENTATION', flags=re.M)
+ANSIBLE_MAJOR = StrictVersion('.'.join(ansible_version.split('.')[:2]))
+
+
+def find_deprecations(o, path=None):
+    if not isinstance(o, (list, dict)):
+        return
+
+    try:
+        items = o.items()
+    except AttributeError:
+        items = enumerate(o)
+
+    for k, v in items:
+        if path is None:
+            this_path = []
+        else:
+            this_path = path[:]
+
+        this_path.append(k)
+
+        if k != 'deprecated':
+            for result in find_deprecations(v, path=this_path):
+                yield result
+        else:
+            try:
+                version = v['version']
+                this_path.append('version')
+            except KeyError:
+                version = v['removed_in']
+                this_path.append('removed_in')
+            if StrictVersion(version) <= ANSIBLE_MAJOR:
+                yield (this_path, version)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'path',
+        help='Path to a directory that will be recursively walked. All .py files will be evaluated '
+             'and any containing DOCUMENTATION will be checked for scheduled deprecations. Default: %(default)s',
+        default=os.path.abspath('ansible/lib'),
+        nargs='?',
+    )
+    args = parser.parse_args()
+
+    plugins = []
+    for root, dirs, filenames in os.walk(args.path):
```

Comment:
I realize this is still a WIP, but I wanted to point out that code-smell tests should use the list of file paths passed in from ansible-test instead of using os.walk.

Reply:
Updated.